src/System.Private.ServiceModel/tests/Scenarios/Client/ExpectedExceptions/ExpectedExceptionTests.4.1.0.cs

```diff
@@ -82,25 +82,54 @@ public static void ServiceRestart_Throws_CommunicationException()
+        // We need the actual host name for the client call to work.
+        // To make it easier to parse, localhost was replaced with '[HOST]'.
+
+        // Use Endpoints.HttpBaseAddress_Basic only for the purpose of extracting the Service host name.
+        // Then update 'restartServiceAddress' with it.
+        UriBuilder builder = new UriBuilder(Endpoints.HttpBaseAddress_Basic);
+        string hostName = builder.Uri.Host;
+        restartServiceAddress = restartServiceAddress.Replace("[HOST]", hostName);
+
+        // Get the last portion of the restart service url which is a Guid and convert it back to a Guid
+        // This is needed by the RestartService operation as a Dictionary key to get the ServiceHost
+        string uniqueIdentifier = restartServiceAddress.Substring(restartServiceAddress.LastIndexOf("/") + 1);
+        Guid guid = new Guid(uniqueIdentifier);
+
+        // *** Step 2 *** \\
+        // Simple echo call to make sure the newly created endpoint is working.
+        try
         {
-            // The restartServiceAddress we got from the Service used localhost as the host name.
-            // We need the actual host name for the client call to work.
-            // To make it easier to parse, localhost was replaced with '[HOST]'.
+            factory = new ChannelFactory<IWcfRestartService>(binding, new EndpointAddress(restartServiceAddress));
+            serviceProxy = factory.CreateChannel();
 
-            // Use Endpoints.HttpBaseAddress_Basic only for the purpose of extracting the Service host name.
-            // Then update 'restartServiceAddress' with it.
-            UriBuilder builder = new UriBuilder(Endpoints.HttpBaseAddress_Basic);
-            string hostName = builder.Uri.Host;
-            restartServiceAddress = restartServiceAddress.Replace("[HOST]", hostName);
+            // *** EXECUTE *** \\
+            string result = serviceProxy.NonRestartService(guid);
 
-            // *** SETUP *** \\
-            // Get the last portion of the restart service url which is a Guid and convert it back to a Guid
-            // This is needed by the RestartService operation as a Dictionary key to get the ServiceHost
-            string uniqueIdentifier = restartServiceAddress.Substring(restartServiceAddress.LastIndexOf("/") + 1);
-            Guid guid = new Guid(uniqueIdentifier);
+            Assert.True(result == "Success!", string.Format("Test Case failed, expected the returned string to be: {0}, instead it was: {1}", "Success!", result));
+        }
+        catch (Exception ex)
+        {
+            string exceptionMessage = ex.Message;
+            string innerExceptionMessage = ex.InnerException.Message;
+            string testExceptionMessage = String.Format("The ping to validate the newly created endpoint failed.\nThe endpoint pinged was: {0}\nThe GUID used to extract the ServiceHost from the server side dictionary was: {1}",restartServiceAddress, guid);
```

Comment:
I like to use the new string coalescing feature as it makes checking what the error message will be easier when the string is long.
```csharp
string testExceptionMessage = $"The ping to validate the newly created endpoint failed.\nThe endpoint pinged was: {restartServiceAddress}\nThe GUID used to extract the ServiceHost from the server side dictionary was: {guid}";
```